api/src/services/product.services.ts

```diff
@@ -25,13 +25,14 @@ const productService = {
   async update(product) {
     const { resource } = await this.container.item(
       product.id,
-      product.brand.name,
+      product.id,
     )
       .replace(product);
     return resource;
   },
-  async delete(id, brandName) {
-    const result = await this.container.item(id, brandName).delete();
+  async delete(id: string) {
+    const result = await this.container.item(id, id).delete();
```

Comment:
Why are we calling twice the `id` here?

Reply:
@juliamuiruri4 what do you think about this `product.services.ts` file? I think this way can be more readable for the audience?

* `product.services.ts` file

```ts
import { CosmosClient } from "@azure/cosmos";

// Set connection string from CONNECTION_STRING value in local.settings.json
const azureCosmosDBconnectionString = process.env.CONNECTION_STRING;

const productService = {

  init() {
    
    try {
      this.client = new CosmosClient(azureCosmosDBconnectionString);
      this.database = this.client.database('tailwind');
      this.container = this.database.container('products');

    } catch (err: unknown) {
      const err = error as Error;
      console.log('Error initializing Cosmos DB client: ', err.message);
      throw err;
    }
  },

  async create(productToCreate) {
    try {
      const { resource } = await this.container.items.create(productToCreate);

      return resource;

    } catch (err: unknown) {
      const err = error as Error;
      console.log('Error creating product: ', err.message);
    }
  },

  async read(): Promise<string> {
    try {
      const iterator = this.container.items.readAll();
      const { resources } = await iterator.fetchAll();

      return JSON.stringify(resources);

    } catch (err: unknown) {
      const err = error as Error;
      console.log('Error reading products: ', err.message);
      throw err;
    }
  },

  async update(product) {
    try {
      const { resource } = await this.container.item(product.id, product.id).replace(product);
      return resource;

    } catch (err: unknown) {
      const err = error as Error;
      console.log('Error reading products: ', err.message);
      throw err;
    }
  },

  async delete(productId: string) {
    try {
      const result = await this.container.item(productId, id).delete();
      return result;

    } catch (err: unknown) {
      const err = error as Error;
      console.log('Error deleting product: ', err.message);
      throw err;
    }
  },
};

// Initialize the productService
productService.init();

export default productService;
```